### PR TITLE
Faeries/echopay upgrade

### PR DESCRIPTION
**Pull request by transfaeries**
this is a different payments demo to use for the walkthrough for the readme.

### CODE REVIEW

File: forest/payments_monitor.py [hunk 1]
```diff
@@ -187,17 +188,51 @@ async def import_account(self) -> dict:
...
+
+        ## get all account IDs for the Wallet / fullservice instance
+        account_ids = (await self.req({"method": "get_all_accounts"}))["result"][
+            "account_ids"
+        ]
+        maybe_account_id = []
+        if account_name is not None:
+            ## get the account map for the accounts in the wallet
+            account_map = [
+                (await self.req({"method": "get_all_accounts"}))["result"][
```
Comment (iamalwaysuncomfortable) [inline forest/payments_monitor.py:213]: A lot of this is addressed in the payments PR but  this seems fine to merge for now, we'll deal with the conflicts once we merge the payments branch

File: echopay/README.md [hunk 2]
```diff
@@ -3,4 +3,18 @@ UNDER CONSTRUCTION
 ```bash
```
Comment (iamalwaysuncomfortable) [inline echopay/README.md:3]: Definitely commit the readme as is. Expansion of it can be a separate PR after this is merged.

**Review by iamalwaysuncomfortable — COMMENTED** [review]
Overall great, a few things to fix but after that's done this should be merged into main. Beyond the comments here let's get mypy, black, and and pylint setup so they pass all those lints!

File: echopay/echopay_final.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import logging
+from unicodedata import decimal
+from urllib import response
+
+from aiohttp import web
+from forest.core import Message, PayBot, Response, app, requires_admin, run_bot
+from forest.utils import get_secret
+from decimal import Decimal
+import mc_util
+from typing import Union
```
[the remark this answers is not in the document]
Comment (technillogue) [reply echopay/echopay_final.py:13]: you can use `isort` for this

[the remark this answers is not in the document]
Comment (iamalwaysuncomfortable) [reply echopay/echopay_final.py:13]: isort could potentially be a pre-commit hook

File: echopay/echopay.py [hunk 4]
```diff
@@ -1,60 +1,91 @@
 #!/usr/bin/python3.9
```
[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay.py:1]: it's a remnant of when my plan was to have the tutorial walk people through modifying the bot, so final, is a sort of "answer key" for users if they get confused. I might not end up using it, but I haven't decided.

[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay.py:1]: but they are meant to be different files

File: echopay/echopay_final.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/python3.9
+from audioop import add
+from inspect import modulesbyfile
```
[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay_final.py:3]: done

File: echopay/echopay_final.py [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+import logging
+from unicodedata import decimal
+from urllib import response
+
+from aiohttp import web
```
[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay_final.py:8]: done

File: echopay/echopay_final.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+from decimal import Decimal
+import mc_util
+from typing import Union
+from forest.payments_monitor import Mobster
+
+
+class Echopay(PayBot):
+
+    # mobster is a class that helps make api calls to the full service API. We use it for account management
+    mobster = Mobster()
```
[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay_final.py:20]: done

File: echopay/echopay_final.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+    fee = int(1e12 * 0.0004)  # Mobilecoin transaction fee
+
+    async def start_process(self) -> None:
+        """Runs when the bot starts and sets the Profile"""
+
+        await self.set_payment_address()
+
+        return await super().start_process()
```
[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay_final.py:18]: changed it to

```python
async def __init__(self) -> None:
        """Creates AND STARTS a bot that routes commands to do_x handlers"""
        super().__init__()
        asyncio.create_task(self.set_payment_address())
```

but the Linter complains `The return type of "__init__" must be None`

[the remark this answers is not in the document]
Comment (transfaeries) [reply echopay/echopay_final.py:18]: nvm, it's because of the async I guess

**Comment by transfaeries** [comment]
I think I've incorporated all the changes and suggestions. Please review again and merge if ready.

**Comment by transfaeries** [comment]
There is one more thing actually, I'll want to adress before merging and I will probably need help doing it. Which is, I need a way to make sure it only sends a message if the payment succeeds and sends a different message to admin and user if it fails. And in a way that's not overly complicated for the reader. So yeah.

**Review by technillogue — APPROVED** [review]
Need to change the names for echopay and echopay_final but that can be done in a later pr

**Review by itdaniher — APPROVED** [review]
Renamed files and added changelog notes. LGTM!
